lib/commands/general.js

```diff
@@ -297,17 +297,24 @@ helpers.suspendChromedriverProxy = function suspendChromedriverProxy () {
 };
 
 commands.startActivity = async function startActivity (appPackage, appActivity,
```

Comment:
is there a documentation for this call that must be updated? AFAIK in clients with strict typing the optionalIntentArguments parameter has type "string" and it is going to be confusing to pass a json there. We either must do the appropriate transformation here or (what I would do) use the corresponding driver-specific `mobile:` extension instead

Reply:
> optionalIntentArguments parameter has type "string" 

side note: 
https://github.com/appium/appium-android-driver/blob/4ec9994570bdf163b47b7318002a2d0b3687f128/lib/desired-caps.js#L132-L134 expects string

Reply:
> corresponding driver-specific `mobile:` extension instead

I wanted to use mobile: startActivity, but is removed in espresso driver and hence passing the intent options as JSON string from client (as client expects and sends String).

But espresso server expects Intent Arguments as Map and hence doing the transformation here from string to JSON.

I can add a new mobile command instead if that is recommended like below

```
commands.mobileStartActivity = async function mobileStartActivity(opts = {}) {
  const {
    appPackage,
    appActivity,
    locale,
    optionalIntentArguments,
    optionalActivityArguments
  } = assertRequiredOptions(opts, ['appPackage', 'appActivity']);
  return await this.espresso.jwproxy.command(`/appium/device/start_activity`, 'POST', {
    appPackage,
    appActivity,
    locale,
    optionalIntentArguments,
    optionalActivityArguments
  });
```

Add `startActivity: 'mobileStartActivity',` to execute.js and will update the doc for expresso mobile:  comments section.

Reply:
having a separate mobile command makes more sense to me, as we could provide a proper documentation for its arguments in driver README. Also, unlike in UIA2, with Espresso we cannot really customize `appPackage` value - it must be equal to the current app under test as Espresso driver only works inside of an instrumented app sandbox.